### PR TITLE
[TASK] Add cross-domain headless links

### LINKED ISSUE #130: [FEATURE] Cross domain links for 2 different sites based on headless
Add possibility to render cross domain links for 2 sites based on headless
@lukaszuznanski will push a POC

### PR DESCRIPTION
**Pull request by lukaszuznanski**
We replace host by frontendHost (frontend url)
Resolves #130

### CODE REVIEW

File: Classes/Routing/PageRouter.php [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+        // cHash is never considered because cHash is built by this very method.
+        unset($originalParameters['cHash']);
+        $enhancers = $this->getEnhancersForPage($pageId, $language);
+        foreach ($enhancers as $enhancer) {
+            if ($enhancer instanceof RoutingEnhancerInterface) {
+                $enhancer->enhanceForGeneration($collection, $originalParameters);
+            }
+        }
+        foreach ($enhancers as $enhancer) {
```
Comment (twoldanski) [inline Classes/Routing/PageRouter.php:127]: You can use one loop with switch for enhacers

Comment (lukaszuznanski) [reply Classes/Routing/PageRouter.php:127]: This is Xclass for existing core method, we don't want to refractor this code, only add new feature

File: Classes/Routing/PageRouter.php [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+     * @return UriInterface
+     * @throws InvalidRouteArgumentsException
+     */
+    public function generateUri($route, array $parameters = [], string $fragment = '', string $type = ''): UriInterface
+    {
+        // Resolve language
+        $language = null;
+        $languageOption = $parameters['_language'] ?? null;
+        unset($parameters['_language']);
+        // Xclass frontendHost
```
Comment (lukaszuznanski) [inline Classes/Routing/PageRouter.php:59]: check if feature switch frontendHost is enabled

**Comment by lukaszuznanski** [comment]
closing, solved by #233

**Pull request closed without merging**